docs/book/getting-started/database-and-models.md

```diff
@@ -208,20 +208,22 @@ To configure the `ServiceManager`, we can either supply the name of the class to
 be instantiated or a factory (closure, callback, or class name of a factory
 class) that instantiates the object when the `ServiceManager` needs it. We start
 by implementing `getServiceConfig()` to provide a factory that creates an
-`AlbumTable`. Add this method to the bottom of the `module/Album/src/Module.php`
+`AlbumTable`. Add the `ServiceProviderInterface` to the class and add this method to the bottom of the `module/Album/src/Module.php`
```

Comment:
The entire module class is too complicated, so it should be reduced to the `getConfig` method.
It makes no sense to add the controller to the configuration file `config/module.config.php` first and then remove it again:

- https://docs.laminas.dev/tutorials/getting-started/modules/#configuration
- https://docs.laminas.dev/tutorials/getting-started/database-and-models/#back-to-the-controller

See also [the section "A Typical Module Class"](https://docs.laminas.dev/laminas-modulemanager/module-class/#a-typical-module-class) in the laminas-modulemanager documentation. (the autoloading is no longer needed!)

I would simplify the whole process for the end user and use the [`ReflectionBasedAbstractFactory` of laminas-servicemanager](https://docs.laminas.dev/laminas-servicemanager/reflection-abstract-factory/). At the end, the factories can be created by [command](https://github.com/laminas/laminas-servicemanager/blob/4.0.x/docs/book/v4/ahead-of-time-factories.md) and replace the `ReflectionBasedAbstractFactory`. The headline could be: "Improving Performance".

Reply:
It would be good if there is one place for the configuration and you don't have to jump between files.

For the class `Album\Model\AlbumTable` the `ReflectionBasedAbstractFactory` can be used and for the class `Album\Model\AlbumTableGateway` a factory should be created as a class, then the benefits described [here](https://docs.laminas.dev/tutorials/getting-started/database-and-models/#factories ) are used from the beginning.

https://docs.laminas.dev/tutorials/getting-started/database-and-models/#using-servicemanager-to-configure-the-table-gateway-and-inject-into-the-albumtable

Reply:
I will simplify the config to use module.config.php.
I agree that using factory classes is a cleaner approach.  I use them myself. 
For the newbie, using the `ReflectionBasedAbstractFactory` may be a little hard to follow unless it is explained properly. 
Is the service manager CLI for AOT only a v4 feature? Has the skeleton been updated to use service-manager v4?

Reply:
> For the newbie, using the ReflectionBasedAbstractFactory may be a little hard to follow unless it is explained properly.

But the handling is much easier and the explain can be:

_The factory class `ReflectionBasedAbstractFactory` reads the type hints of the constructor and automatically passes the correct services to the constructor. This process is also known as "autowiring"._ 

> Is the service manager CLI for AOT only a v4 feature? Has the skeleton been updated to use service-manager v4?

Not yet, but the "Performance Improvement" section can also be added later.

Reply:
OK.  I will update accordingly.

Reply:
Change as much as you want here, further changes can be made in other pull requests.

Reply:
I will update the Getting Started section so that it is complete.
The other sections need updating as well and I will do them separately